bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/util/ModelUtil.java

```diff
@@ -138,24 +141,32 @@ public static void verifyLinkIntermediateEvents(
     // get link throw events definition
     final List<EventDefinition> linkThrowEvents = getEventDefinitionsForLinkThrowEvents(element);
 
-    // get link catch events names
-    final Set<String> linkCatchList =
+    // link catch events names group by scope
+    final Map<BpmnModelElementInstance, Set<String>> linkCatchEventsGroupByScope =
         getEventDefinition(linkCatchEvents, LinkEventDefinition.class)
             .filter(def -> def.getName() != null && !def.getName().isEmpty())
-            .map(LinkEventDefinition::getName)
-            .collect(Collectors.toSet());
+            .collect(
+                groupingBy(
+                    LinkEventDefinition::getScope,
+                    mapping(LinkEventDefinition::getName, Collectors.toSet())));
```

Comment:
🔧 You can extract this method, like: 

```
    private Map<BpmnModelElementInstance, Set<String>> groupEventsByScope(List<LinkEventDefinition> events) {
        return events.stream()
            .filter(def -> def.getName() != null && !def.getName().isEmpty())
            .collect(Collectors.groupingBy(
                LinkEventDefinition::getScope,
                Collectors.mapping(LinkEventDefinition::getName, Collectors.toSet())
            ));
    }
```

Then, you can use the extracted method like this:
```
Map<BpmnModelElementInstance, Set<String>> linkCatchEventsGroupByScope = groupEventsByScope(linkCatchEvents);
Map<BpmnModelElementInstance, Set<String>> linkThrowEventsGroupByScope = groupEventsByScope(linkThrowEvents);
```

By doing this we remove code duplication and improve readability 👀